Newspack/Newspack/Stores/FolderStore.swift

```diff
@@ -245,39 +217,29 @@ extension FolderStore {
         }
     }
 
-    /// Rename a story folder. This updates the name of the story folder's underlying
-    /// directory as well as the name field in core data.
-    ///
-    /// - Parameters:
-    ///   - uuid: The uuid of the StoryFolder to update.
-    ///   - name: The new name.
-    ///
-    func renameStoryFolder(uuid: UUID, to name: String) {
-        // Get the folder.
-        guard let storyFolder = getStoryFolderByID(uuid: uuid) else {
-            LogError(message: "Unable to find the story folder to rename.")
-            return
-        }
+}
 
-        // Update the name of its folder. We will assume it is not stale.
+// MARK: - Story Folder Selection
+
+extension FolderStore {
+
+    /// Select the default story folder if needed.
+    ///
+    private func selectDefaultStoryFolderIfNeeded() {
+        // Make sure there are story folders to retrieve.
         guard
-            let url = folderManager.urlFromBookmark(bookmark: storyFolder.bookmark),
-            let newUrl = folderManager.renameFolder(at: url, to: name)
+            let _ = currentSiteID,
+            getStoryFolderByID(uuid: currentStoryFolderID) == nil,
+            let firstFolder = getStoryFolders().first
         else {
-            LogError(message: "Unable to rename story folder")
             return
         }
-
-        LogInfo(message: "Success: \(newUrl)")
-
-        // Save the name in core data.
-        let objID = storyFolder.objectID
-        CoreDataManager.shared.performOnWriteContext { context in
-            let folder = context.object(with: objID) as! StoryFolder
-            folder.name = name
-
-            CoreDataManager.shared.saveContext(context: context)
+        // Check for last selected story folder.
+        if let lastFolder = getLastSelectedStoryFolder() {
+            selectStoryFolder(uuid: lastFolder.uuid)
+            return
         }
+        selectStoryFolder(uuid: firstFolder.uuid)
```

Comment:
Nipticky!! You can simplify this snippet:

```
let folder = getLastSelectedStoryFolder() ?? firstFolder
selectStoryFolder(uuid: folder.uuid)
```

Reply:
Nice! I like it :)